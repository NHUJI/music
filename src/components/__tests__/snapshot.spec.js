import { shallowMount, RouterLinkStub } from "@vue/test-utils";
import SongItem from "@/components/SongItem.vue";
import { expect, test } from "vitest";

describe("Snapshots SongItem.vue", () => {
  test("renders correctly", () => {
    const song = {
      docID: "abc",
      modified_name: "test",
      display_name: "test",
      comment_count: 5,
    };

    const wrapper = shallowMount(SongItem, {
      propsData: { song },
      global: {
        components: {
          "router-link": RouterLinkStub,
        },
      },
    });

    // 生成快照(没有时生成，有时和之前生成的比较)
    expect(wrapper.element).toMatchSnapshot();
  });
});
