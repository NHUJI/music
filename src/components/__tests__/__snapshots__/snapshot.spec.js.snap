// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshots SongItem.vue > renders correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50 example snapshot song-id-abc"
>
  <!-- 添加了动态属性id作为测试 -->
  <div>
    <!-- name and id must provide to the router as params to replace the dynamic segment -->
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    >
      test
    </span>
  </div>
  <div
    class="text-gray-60 0 text-lg"
  >
    <!-- use custom property to complete control over the behavior of the component  -->
    <router-link-stub
      custom="true"
      to="[object Object]"
    />
  </div>
</li>
`;
